core/modules/replica/Application.cc

```diff
@@ -156,6 +172,11 @@ int Application::run() {
         _serviceProvider = ServiceProvider::create(_config, _instanceId);
         _serviceProvider->run();
     }
+    // Change default parameters of the XROOTD connectors
+    if (_injectXrootdOptions) {
+        Configuration::setXrootdAllowReconnect(_xrootdAllowReconnect != 0);
```

Comment:
This requires some explanation on how the simple class `Parser` works. This class is defined in header `ApplicationTypes.h` of the same Qserv module `replica`. The parser allows:
* positional parameters like `<p1> <p2> ...`, where optional positional parameters are also allowed like `<p1> <p2> [<p3> [<p4> [<p5 ...]]]`
* options having values like `--opt1=<val> --opt2=<val>`. Boolean values aren't allowed as `<val>`. They would be treated as strings.
* and flags like `--verbose`, etc. If a flag is not present then the corresponding variable would stay as `false`. f the flag is found in a command line then the variable would flip to `true`.

The implementation of class `Application` tries some non-trivial actions:
```
// ---------------
// Application.h
unsigned int _xrootdAllowReconnect;

// ---------------
// Application.cc
// First of all it pulls the default value of the flag from the `Configuration`, so that a user of the application
// could see the default value by passing flag in the command line:
//     --help
// Note that the default value inside class Configuration is implementation-specific, and it may change.
_xrootdAllowReconnect = Configuration::xrootdAllowReconnect() ? 1 :0;

// On the next step the parser allow option:
//   --xrootd-allow-reconnect=<value>
// Where the default value of the parameter is set at the previous step.
parser().option(
    "xrootd-allow-reconnect",
    "Change the default XRootD/SSI connection handling node. Set 0 to disable"
    " automatic reconnects. Any other number would allow reconnects.",
    _xrootdAllowReconnect
);

After the parsing is done then the class `Application` would update the value inside `Configuration`.
Configuration::setXrootdAllowReconnect(_xrootdAllowReconnect !=0);
```
This algorithm works equally well regardless of the initial value of this parameter inside the class `Configuration.
In case if I wanted to define the variable as:
```
bool _xrootdAllowReconnect;
_xrootdAllowReconnect = Configuration::xrootdAllowReconnect();
```
Then I would run into a little logical problem that would require me to introduce two flags, one to reset, ad the other one to set the flag:
```
--xrootd-allow-reconnect
--no-xrootd-allow-reconnect
```
The integer representation of the flag inside `Application` solved that dilemma.